### DISCUSSION TITLE
Cannot add lint-staged: only eslint, stylelint, prettier or custom rules are supported. Continuation of discussion.

### DISCUSSION THREAD

**Discussion by vogelino** [sapegin/mrm#323] [Bugs]
I attempted to continue the discussion on #147 but the owner gently just shot down the issue and blocked all comments. What a way to treat your community. Feel free to do the same here but I just wanted to add:

I was able to identify the problem. The error message suggests that only some libraries are supported, but the error in my case was that eslint wasn't installed as devDependencies but instead globally. This is stated is the docs, but the error message could be made more helpful with an addition such as "Are you sure that you installed all dependencies locally"?

**Comment by VladLegkowski** [comment]
Same issue happens on the CRA with pretty simple package.json, I am confused!

**Comment by bbugh** [comment]
I also had this issue setting it up on a monorepo, which has multiple different languages and will have *all* custom scripts. I think it should warn and then install a minimal setup instead of just failing.

**Comment by alex-grover** [comment]
To be 100% clear for future readers, `eslint` and `prettier` specifically have to be in `devDependencies`. Regular `dependencies` does not work, having them installed as sub-dependencies, peer dependencies, or globally does not work.

The task tries to determine what tools you're using not just based on your npm scripts [as implied by the documentation](https://mrm.js.org/docs/mrm-task-lint-staged#what-it-does). You can find the code [here](https://github.com/sapegin/mrm/blob/master/packages/mrm-task-lint-staged/index.js#L27-L55).

It is definitely confusing that the task has pretty specific requirements for what it's looking for, especially if you're coming from the `lint-staged` readme (like i'm sure many people are) that doesn't indicate anything besides "make sure you have your other tools set up". The error message is very unclear and at the very least updating that to indicate to the user what configuration it's looking for would be helpful

**Comment by cengit** [comment]
before running **npx mrm@2 lint-staged**
make sure you have eslint and prettier in you devDependencies of package.json
```json
{
  "eslint": "^some.version.number",
    "prettier": "^some.version.number"
}
```

**Comment by millette** [comment]
Could [dprint](https://www.npmjs.com/package/dprint) be supported directly?
